tests/test_utilities.cpp

```diff
@@ -111,37 +111,31 @@ std::string moduleName<ThermalConduction>()
  * @brief Verifies the solution fields against the input file
  * @param[in] phys_module The module whose fields should be verified
  * @param[in] inlet The Inlet object from which expected solution values will be obtained
- * @param[in] dim The mesh dimension
  */
 template <typename PhysicsModule>
-void verifyFields(const PhysicsModule&, const axom::inlet::Inlet&, const int)
+void verifyFields(const PhysicsModule&, const axom::inlet::Inlet&)
 {
   static_assert(AlwaysFalse<PhysicsModule>::value, "Test driver is not supported for selected type");
 }
 
 template <>
-void verifyFields(const Solid& phys_module, const axom::inlet::Inlet& inlet, const int dim)
+void verifyFields(const Solid& phys_module, const axom::inlet::Inlet& inlet)
 {
-  mfem::Vector zero(dim);
-  zero = 0.0;
-  mfem::VectorConstantCoefficient zerovec(zero);
-
   if (inlet.contains("expected_u_l2norm")) {
-    double x_norm = phys_module.displacement().gridFunc().ComputeLpError(2.0, zerovec);
+    double x_norm = phys_module.displacement().norm();
     EXPECT_NEAR(inlet["expected_u_l2norm"], x_norm, inlet["epsilon"]);
   }
   if (inlet.contains("expected_v_l2norm")) {
-    double v_norm = phys_module.velocity().gridFunc().ComputeLpError(2.0, zerovec);
+    double v_norm = phys_module.velocity().norm();
     EXPECT_NEAR(inlet["expected_v_l2norm"], v_norm, inlet["epsilon"]);
   }
 }
 
 template <>
-void verifyFields(const ThermalConduction& phys_module, const axom::inlet::Inlet& inlet, const int)
+void verifyFields(const ThermalConduction& phys_module, const axom::inlet::Inlet& inlet)
 {
-  mfem::ConstantCoefficient zero(0.0);
   if (inlet.contains("expected_t_l2norm")) {
-    double t_norm = phys_module.temperature().gridFunc().ComputeLpError(2.0, zero);
+    double t_norm = phys_module.temperature().norm();
     EXPECT_NEAR(inlet["expected_t_l2norm"], t_norm, inlet["epsilon"]);
   }
```

Comment:
could this be equivalently implemented as 
```cpp
template <typename PhysicsModule>
void verifyFields(const PhysicsModule&, const axom::inlet::Inlet&) {
  if (inlet.contains("expected_u_l2norm")) {
    EXPECT_NEAR(inlet["expected_u_l2norm"], phys_module.displacement().norm(), inlet["epsilon"]);
  }
  if (inlet.contains("expected_v_l2norm")) {
    EXPECT_NEAR(inlet["expected_v_l2norm"], phys_module.velocity().norm(), inlet["epsilon"]);
  }
  if (inlet.contains("expected_t_l2norm")) {
    EXPECT_NEAR(inlet["expected_t_l2norm"], phys_module.temperature().norm(), inlet["epsilon"]);
  }
}
```
If so, then future multiphysics solvers (e.g. ThermalSolid) would likely already work without requiring additional overloads.

On an unrelated note, I think a lowercase "t" is an unconventional symbol for temperature, is it worth renaming that field in inlet? With user-facing interfaces I prefer more verbose names, maybe
`expected_displacement_l2norm`,  `expected_velocity_l2norm`, `expected_temperature_l2norm`.

Reply:
I think it's kind of nice to keep these as separate instantiations to guard against physics modules that are not supported. I agree with your second point though, so I'll open an issue.